### PR TITLE
fix(llmobs): recreate eval metric writer on fork

### PR DESCRIPTION
**Pull request by lievan**
This PR ensures that the `LLMObsEvalMetricWriter` is correctly recreated and restarted on a forked process.

Previously, on a process fork we were not recreating/restarting the eval metric writer worker.

Mirrors https://github.com/DataDog/dd-trace-py/pull/10249 but for eval metric writer

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/llmobs/test_llmobs_service.py [hunk 1]
```diff
@@ -1403,20 +1403,7 @@ def test_activate_distributed_headers_activates_context(LLMObs, mock_logs):
             mock_activate.assert_called_once_with(dummy_context)
 
 
-def _task(llmobs_service, errors, original_pid, original_span_writer_id):
```
Comment (lievan) [inline tests/llmobs/test_llmobs_service.py:1406]: i don't think this is used anywhere, cool if we remove? @Yun-Kim

Comment (Yun-Kim) [reply tests/llmobs/test_llmobs_service.py:1406]: Yes, thanks for the catch

File: tests/llmobs/test_llmobs_service.py [hunk 2]
```diff
@@ -1470,6 +1481,37 @@ def test_llmobs_fork_create_span(monkeypatch):
         llmobs_service.disable()
 
 
+def test_llmobs_fork_submit_evaluation(monkeypatch):
+    """Test that forking a process correctly encodes new spans created in each process."""
+    monkeypatch.setenv("_DD_LLMOBS_WRITER_INTERVAL", 5.0)
+    with mock.patch("ddtrace.internal.writer.HTTPWriter._send_payload"):
```
Comment (Yun-Kim) [inline tests/llmobs/test_llmobs_service.py:1487]: The eval metric writer does not extend from `HTTPWriter`, we'll need to use a different mock for this

**Review by emmettbutler — COMMENTED** [review]
I'll wait to review until existing comments are addressed and tests are passing. Please re-request a review from me at that time. Thanks!

File: ddtrace/llmobs/_writer.py [hunk 3]
```diff
@@ -141,6 +141,15 @@ def _url(self) -> str:
     def _data(self, events: List[Any]) -> Dict[str, Any]:
         raise NotImplementedError
 
+    def recreate(self):
+        # type: () -> BaseLLMObsWriter
```
Comment (Yun-Kim) [inline ddtrace/llmobs/_writer.py:145]: Type hinting is directly in the function signature now
```suggestion
    def recreate(self) -> BaseLLMObsWriter:
```

File: releasenotes/notes/fix-llmobs-forked-eval-metric-writer-55d50d1ed8fc10ea.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+fixes:
+  - |
+    LLM Observability: This fix resolves an issue where LLM Observability evaluation metrics were not being submitted 
+    in forked processes, The LLM Observability evaluation metric writer thread now automatically restarts  when a forked
```
Comment (Yun-Kim) [inline releasenotes/notes/fix-llmobs-forked-eval-metric-writer-55d50d1ed8fc10ea.yaml:5]:
```suggestion
    in forked processes. The evaluation metric writer thread now automatically restarts when a forked 
```

**Review by Yun-Kim — APPROVED** [review]
Minor nits, but LGTM

**Pull request merged**